Add relative difference from mean to depth statistics

scw opened the pull request:
This is a recommended statistic from Lecours 2017 paper, and should be included in the basket of depth statistics we include.

scw committed:
pep8 cleanup

scw committed:
allow more memory for calculations in 64-bit

If we're in a 64-bit environment, either background geoprocessing is
enabled, we're in a Server context, or we're in Pro. In all three cases,
we can be less conservative with memory.

scw committed:
Add relative difference to mean calculation

scw committed:
expose difference to mean to tools

noahslocum commented on Install/toolbox/scripts/depth_statistics.py:
```diff
@@ -121,6 +122,22 @@ def main(in_raster=None, neighborhood_size=None,
                 utils.msg("saving mean depth to {}".format(mean_raster))
             arcpy.CopyRaster_management(mean_depth, mean_raster)
 
+            if diff_mean_set.intersection(out_stats):
```
It looks like if a user selects `Difference to Mean` but not `Mean Depth`, the tool will not produce any results. Can we change this to calculate `Mean Depth` either way, so that `Difference to Mean` can be calculated alone?

scw replied:
Yup, great catch! Let me fix this and resubmit.

noahslocum reviewed (changes requested):
Sounds like a valuable statistic to have! When I ran this the tool did not seem to ever get to the `Difference to Mean` logic though:
![image](https://user-images.githubusercontent.com/19574742/31784319-1aade92c-b4b6-11e7-8f8a-0fa97eb874c2.png)

I tried to debug a bit but did not see anything obviously wrong. Any ideas?

scw committed:
add test.

scw committed:
add window_type to both tools.

Add it to the end of the function call, but keep it in a place that
makes sense in the parameter display order.

scw committed:
Add neighborhood type calculation

scw committed:
break out naming into functions to DRY out this

scw committed:
final changes to sync up the set names.

scw committed:
update documentation.

 - add new window type parameter.
 - add new 'relative difference to mean' calculation.

scw commented:
@noahslocum OK, I've submitted new commits which should fix the issues. Thanks!

scw committed:
include window type in error

scw committed:
modify parameters based on neighborhood type

scw committed:
remove exec usage

scw committed:
pep8 fixes

scw committed:
set default NoData value if missing

noahslocum commented:
With the recent changes this new stat seems to be working fine, sorry for taking so long to approve!
